extensions/cornerstone/src/services/ViewportService/CornerstoneViewportService.ts

```diff
@@ -548,115 +568,153 @@ class CornerstoneViewportService extends PubSubService
     const displaySetInstanceUIDs = this.viewportsDisplaySets.get(viewport.id);
 
     const segDisplaySet = displaySetInstanceUIDs
-      .map(displaySetService.getDisplaySetByUID)
+      .map(DisplaySetService.getDisplaySetByUID)
       .find(displaySet => displaySet && displaySet.Modality === 'SEG');
```

Comment:
Please add something about the display set which indicates that this is an "overlay" display set of some sort rather than embedding knowledge here about segmentations and rt struct, so that that this relies on something coming from the getSOPClassHandler, eg:
const overlayDisplaySet = displaySetInstanceUIDs.map(displaySetService.getDisplaySetByUID)
.find(displaySet => displaySet?.addSegmentationHandler)